lib/Runtime/PlatformAgnostic/Platform/Common/Intl.cpp

```diff
@@ -482,14 +516,146 @@ namespace Intl
         error = UErrorCode::U_ZERO_ERROR;
 
         written = uloc_toLanguageTag(defaultLocaleId, bcp47, ULOC_FULLNAME_CAPACITY, true, &error);
-        ICU_ASSERT_RETURN(error, written > 0 && written < cchLangtag, 0);
+        ICU_ASSERT_RETURN(error, written  > 0 && ((size_t) written) < cchLangtag, 0);
```

Comment:
>(size_t)  [](start = 50, length = 9)

nit: can we use static_cast or something?

Reply:
I thought it made sense to not static_cast here since it added a lot of line length and we know its safe because we have already asserted its greater than 0.

Reply:
Fair enough. Not a big deal either way.